src/versionize/src/primitives.rs

```diff
@@ -361,6 +361,143 @@ mod tests {
         }
     }
 
+    #[derive(Debug, serde_derive::Deserialize, PartialEq, serde_derive::Serialize, Versionize)]
+    enum CompatibleEnum {
+        A,
+        B(String),
+        C(u64, u64, char),
+    }
+
+    #[derive(Debug, serde_derive::Deserialize, serde_derive::Serialize, Versionize)]
+    struct TestCompatibility {
+        _string: String,
+        _array: [u8; 32],
+        _u8: u8,
+        _u16: u16,
+        _u32: u32,
+        _u64: u64,
+        _i8: i8,
+        _i16: i16,
+        _i32: i32,
+        _i64: i64,
+        _f32: f32,
+        _f64: f64,
+        _usize: usize,
+        _isize: isize,
+        _vec: Vec<u64>,
+        _option: Option<bool>,
+        _enums: Vec<CompatibleEnum>,
+        _box: Box<String>,
+    }
+
+    #[test]
+    fn test_bincode_deserialize_from_versionize() {
+        let mut snapshot_mem = vec![0u8; 4096];
+        let vm = VersionMap::new();
+
+        let test_struct = TestCompatibility {
+            _string: "String".to_owned(),
+            _array: [128u8; 32],
+            _u8: 1,
+            _u16: 32000,
+            _u32: 0x1234_5678,
+            _u64: 0x1234_5678_9875_4321,
+            _i8: -1,
+            _i16: -32000,
+            _i32: -0x1234_5678,
+            _i64: -0x1234_5678_9875_4321,
+            _usize: 0x1234_5678_9875_4321,
+            _isize: -0x1234_5678_9875_4321,
+            _f32: 0.123,
+            _f64: 0.123_456_789_000_000,
+            _vec: vec![33; 32],
+            _option: Some(true),
+            _enums: vec![
+                CompatibleEnum::A,
+                CompatibleEnum::B("abcd".to_owned()),
+                CompatibleEnum::C(1, 2, 'a'),
+            ],
+            _box: Box::new("Box".to_owned()),
+        };
+
+        Versionize::serialize(&test_struct, &mut snapshot_mem.as_mut_slice(), &vm, 1).unwrap();
+
+        let restored_state: TestCompatibility =
+            bincode::deserialize_from(snapshot_mem.as_slice()).unwrap();
+        assert_eq!(test_struct._string, restored_state._string);
```

Comment:
Having an expanded assertion simplifies debugging by showing which primitive is broken.

Reply:
nit: I like the `PartialEq` solution better too; if something's broken, someone will look into it and figure it out eventually.

Reply:
After more thought on this I've also convinced myself:) I'll need to move out the enum and derive the `PartialEq` for the big struct. That would mean 2 asserts instead of many.